client/state/posts/test/utils.js

```diff
@@ -16,10 +16,80 @@ import {
 	getDeserializedPostsQueryDetails,
 	getSerializedPostsQueryWithoutPage,
 	getTermIdsFromEdits,
+	hasTermEditDifferences,
 	mergeIgnoringArrays
 } from '../utils';
 
 describe( 'utils', () => {
+	describe( 'hasTermEditDifferences', () => {
+		it( 'should return false if no term edits passed', () => {
+			expect( hasTermEditDifferences() ).to.be.false;
+		} );
+
+		it( 'should return false if term edits are the same as saved terms', () => {
+			const hasDifferences = hasTermEditDifferences( {
+				post_tag: [ 'ribs', 'chicken' ],
```

Comment:
When the state is updated via the term selector, or token field.  But there are a few different normalization flows that pertain to editing and transforming data in terms prior to sending to the API https://github.com/Automattic/wp-calypso/blob/master/client/state/posts/utils.js#L40-L46